core/sql/parser/sqlparser.y

```diff
@@ -19430,6 +19432,124 @@ exists_predicate : TOK_EXISTS rel_subquery
 				  $$ = new (PARSERHEAP()) Exists($2);
 				}
 
+overlaps_predicate : value_expression_list_paren TOK_OVERLAPS value_expression_list_paren
+         {
+            ItemExprList  exprList1($1, PARSERHEAP());
+            ItemExprList  exprList2($3, PARSERHEAP());
+            //Syntax Rules: 
+            //  1) The degrees of <row value predicand 1> and <row value predicand 2> shall both be 2.
+            if ((exprList1.entries() != 2)
+                  || (exprList1.entries() != exprList2.entries()))
+              {
+                 *SqlParser_Diags << DgSqlCode(-4077)
+                     << DgString0("OVERLAPS");
+                 YYERROR;  // CHANGE TO YYABORT
+              }
+
+            ItemExpr *D1 = (*$1)[0];
+            ItemExpr *E1 = (*$1)[1];
+            ItemExpr *D2 = (*$3)[0];
+            ItemExpr *E2 = (*$3)[1];
+
+            NABoolean negate = FALSE;
+            ItemExpr *tmpItemExpr = E1;
```

Comment:
This logic should be in the Binder, not in the Parser. The Parser does not know the datatypes of anything except constants. And I don't think you want to limit OVERLAPS to operating only on constants.

Reply:
Yes, @DaveBirdsall,  Thanks for your reminding, i will changed this.